litex/build/parser.py

```diff
@@ -217,13 +217,21 @@ def parse_args(self, args=None, namespace=None):
                 self._platform.fill_args(self._toolchain, self)
 
         # Intercept selected CPU to fill arguments.
-        cpu_cls = cpu.CPUS.get(self.get_value_from_key("--cpu-type"), None)
+        default_cpu_type = self._args_default.get("cpu_type", None)
+        cpu_cls = cpu.CPUS.get(self.get_value_from_key("--cpu-type", default_cpu_type))
         if cpu_cls is not None and hasattr(cpu_cls, "args_fill"):
             cpu_cls.args_fill(self)
 
         # Injects arguments default values
         if len(self._args_default):
             argparse.ArgumentParser.set_defaults(self, **self._args_default)
+            # Catch defaults which do not match any arguments - typos?
+            remaining = list(self._args_default.keys())
+            for action in self._actions:
+                if action.dest in remaining:
+                    remaining.remove(action.dest)
+            if len(remaining) > 0:
+                raise ValueError(f"set_default() for invalid argument(s): {remaining}")
```

Comment:
Having a test and/or a message looks good, but raising an exception may, potentially being a problem for users: 
- if in a target I set `cpu_type=vexriscv_smp` and `dtlb_size=xxx`
- if a user decide to use `--cpu-type=naxriscv`

An exception will occur because `dtlb_size` is something specific to the `vexriscv_smp`.

Maybe it's better here to displays a message (similar to [compat_notice](https://github.com/enjoy-digital/litex/blob/master/litex/compat/__init__.py#L6)) with unused/unknown arguments list with and a delay to warn user about situation. Otherwise user has to modifiy target to remove/comment default and it's not really user friendly.